### ISSUE TITLE
Small delay option for users who can see (a bit)

### ISSUE THREAD

**Issue by nozem79** [nvaccess/nvda#11898]
A way to get a small delay of 0.5 - 1 seconds before NVDA starts speaking, that would be of a great help for people who can see a bit or have dyslexia (or like me, have aphasia).
Its a bit like Claroread works

**Comment by leonardder** [comment]
Honestly, I do not understand how this can benefit anyone.

**Comment by nozem79** [comment]
Form my PC and Mac I use Claroread, Claroread has a small delay, so if people use the mouse, not everyting it touches starts speaking, only when the mouse stops .

For dyslectic people this is a must, becouse they only whant to hear where the mouse halts, not everything inbetween...

**Comment by nozem79** [comment]
I changed the status, so the questions is more clear

**Comment by ABuffEr** [comment]
Hi,
according to the implementation, there might be a little advantage even for totally blind users (like me), in some specific situation. I think, particularly, when NVDA starts speaking when audio card (or external trasmitter connected to it) is in stand-by, and the first second or so of the spoken message goes lost.
My 2 cents :)

**Comment by nozem79** [comment]
Yay, maybe I'l look for a developer so I can get it as a add-on.

**Comment by Adriani90** [comment]
If you set the Navigator to follow the mouse cursor in object presentation settings, then you can always press nvda+shift+o to make NVDA report the object which is currently focused under the mouse.

Closing as works for me.

**Comment by hwf1324** [comment]
> If you set the Navigator to follow the mouse cursor in object presentation settings, then you can always press nvda+shift+o to make NVDA report the object which is currently focused under the mouse.
> 
> Closing as works for me.

No, the request for this issue is to delay reporting what is under the mouse, not to manually view what is under the mouse.

**Comment by Adriani90** [comment]
The workaround is comfortable enough.Von meinem iPhone gesendetAm 06.07.2024 um 02:28 schrieb WangFeng Huang ***@***.***>:﻿

If you set the Navigator to follow the mouse cursor in object presentation settings, then you can always press nvda+shift+o to make NVDA report the object which is currently focused under the mouse.
Closing as works for me.

No, the request for this issue is to delay reporting what is under the mouse, not to manually view what is under the mouse.

**Comment by hwf1324** [comment]
The actual need is to not report content in between, rather than subsequently viewing it manually.

**Comment by CyrilleB79** [comment]
The workaround suggested by @Adriani90 does not answer to the initial request, at least, not obviously without any approval from @nozem79. Re-opening.


Reopening

**Comment by Adriani90** [comment]
I think a way to solve this would be a point and click feature as a new mouse navigation mode in NVDA. When the user needs to hear what's under the mouse, then he or she could press a button on the mouse and NVDA speaks the currently focused element.
That means for that mouse navigation mode the mouse click event is overwritten by NVDA and the standard double click would become a tripple click, while the single click would become a double click.

**Comment by SaschaCowley** [comment]
@Adriani90 why do you believe that is preferable to @nozem79's original suggestion?

**Comment by Adriani90** [comment]
@SaschaCowley 
1. The original proposal might introduce the impression that the speech performance is bad
2. The original performance does not solve the issue where people need time to process information, e.g. when a popup notification / or an alert appears and interupts the previous speech
3. If people that need this feature can see a bit, I think they still rely on visual elements, and so they don't need speech every time on every element.
4. Having a point and click speech feature would make the experience very teylor made, and the user can decide when to click. The user can try to read the information with the eyes first, and click on the element when speech is needed.

If a point and click speech feature is implemented, it should inform the user that the mouse gesture of left click changed as described in my last comment above, when starting NVDA with this feature enabled.

**Comment by seanbudd** [comment]
We should address this by adding an unbound command to read whats under the mouse. Once #19495 we can a bound gesture using mouse click

**Comment by CyrilleB79** [comment]
This feature (original request) has just [been requested](https://groups.io/g/nvda-fr/topic/proposition_de/119624637) by a French visually impaired user who has just discovered the new Magnifier feature in beta.
He writes (translated from French):
> The ability to set a speech delay when moving the mouse pointer when mouse tracking is enabled. This would allow a visually impaired user to move their pointer without NVDA speaking at the slightest movement. On the other hand, NVDA could announce the element located under the pointer as soon as the pointer remains on the element for a few milliseconds. For information, this possibility exists with Zoomtext and it is a significant convenience.

I'm Would not be a user of this feature myself. But my opinion is that the initial feature request is not the same as the "Click and read" one, suggested later by @Adriani90 and @seanbudd.
The difference is the same as, with keyboard, having focus reported automatically or on-demand with `NVDA+tab`: both are useful but not in the same situations.

**Comment by nozem79** [comment]
Still a little bit silly it's no option yet, another smart option would be a function that when enabled will stop speaking when the mouse is moving at all.

That would half half impaired, dyslexic people or issues like me, aphasia...

In the meantime, I spent my time on building an app  on Linux, so I don't have to use NVDA anymore, since I made my own app, I can use my favorite operating system which I could not use since my accident 15 years ago :-)



Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by CyrilleB79** [comment]
> another smart option would be a function that when enabled will stop speaking when the mouse is moving at all.

This one already exists: in mouse settings dialog, uncheck "Enable mouse tracking". Or just press NVDA+M to toggle this option without having to open mouse settings dialog.

**Comment by nozem79** [comment]
Oh, good to know, I midded that :-) Now I can make an USB stick with TTS for Windows & Linux

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by hwf1324** [comment]
I have experimentally implemented this feature in version 0.9.0 of the Mouse Enhancements add-on.

Please note that this relies on the automatic update mouse object feature, which requires setting the update method to core cycle.

Due to the vision framework, the control for setting the mouse movement event delay is a slider. Using a SpinCtrl would be difficult.
